lib/api/contracts/model_contract.rb

```diff
@@ -51,6 +51,27 @@ def self.attribute(*attributes, &block)
       validate :readonly_attributes_unchanged
       validate :run_attribute_validations
 
+      # Performs validation of both, the contract and the underlying model and returns
+      # the merged validation errors for both of them.
+      def validate_contract_and_model
```

Comment:
Overriding is not an option as I also change the semantics (which is not even reflected in the new name).

`validate` performs validation, fills the `errors` and returns whether or not validation succeeded.

This method performs validation for both, contract and model and returns **the merged errors**, instead of a truthy/falsey value.

I wanted to depart from the practice of "polluting" the `errors` of either object with entries from the other one, that is why I do the duplication business. But this is also the reason I need to return the merged result immediately.

TBH: I already feel bad for calling the method `validate_x`, because you might not expect the return type I give to you. Which is more problematic in a dynamic language, because it is all about the expectations, isn't it? ^^